src/interfaces/IPositionManager.sol

```diff
@@ -12,11 +12,20 @@ interface IPositionManager is INotifier {
     error DeadlinePassed();
     error IncorrectPositionConfigForTokenId(uint256 tokenId);
 
-    /// @notice Batches many liquidity modification calls to pool manager
+    /// @notice Unlocks Uniswap v4 PoolManager and batches actions for modifying liquidity
+    /// @dev This is the standard entrypoint for the PositionManager
     /// @param payload is an encoding of actions, and parameters for those actions
     /// @param deadline is the deadline for the batched actions to be executed
     function modifyLiquidities(bytes calldata payload, uint256 deadline) external payable;
 
+    /// @notice Batches actions for modifying liquidity without unlocking v4 PoolManager
+    /// @dev This must be called by a contract that has already unlocked the v4 PoolManager
+    /// @param actions the actions to perform
+    /// @param params the parameters to provide for the actions
+    function modifyLiquiditiesWithoutUnlock(bytes calldata actions, bytes[] calldata params) external payable;
```

Comment:
should this also have a deadline parameter?

Reply:
thought about it a little bit, and i dont think so

this call is made from a contract (since PM is already unlocked), so callers (i.e. hooks) should enforce deadlines

deadline on `modifyLiquidities` makes sense because clients / wallets are interacting with it directly and wont be able to enforce deadline